loongarch64-unknown-linux-musl target does not build

```
cargo +nightly build -Zbuild-std --target=loongarch64-unknown-linux-musl
```
```
rustc 1.78.0-nightly (2d24fe591 2024-03-09)
binary: rustc
commit-hash: 2d24fe591f30386d6d5fc2bb941c78d7266bf10f
commit-date: 2024-03-09
host: x86_64-unknown-linux-gnu
release: 1.78.0-nightly
LLVM version: 18.1.0
```

The output is miles of errors like this:
```
error[E0412]: cannot find type `c_char` in the crate root
  --> /home/ben/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.153/src/unix/mod.rs:52:29
   |
6  | pub type c_schar = i8;
   | ---------------------- similarly named type alias `c_schar` defined here
...
52 |         pub gr_name: *mut ::c_char,
   |                             ^^^^^^
   |
help: a type alias with a similar name exists
   |
52 |         pub gr_name: *mut ::c_schar,
   |                             ~~~~~~~
help: consider importing this type alias
   |
155+         use ffi::c_char;
   |
help: if you import `c_char`, refer to it directly
   |
52 -         pub gr_name: *mut ::c_char,
52 +         pub gr_name: *mut c_char,
   |
```

I think this target has somehow slipped past our CI? Maybe I'm searching our codebase wrong, but I don't think I can find any tests for it? Note that this affects the `-musl` target specifically, as identified by the Miri cron job that tries to build all tier 2 targets: https://github.com/rust-lang/miri/blob/66cf3418bd5f9de9a726681c054b3852096d6a68/.github/workflows/sysroots.yml which just failed: https://github.com/rust-lang/miri/actions/runs/8219608617/job/22477773992

cc target maintainers: @heiher @xiangzhai @zhaixiaojuan @xen0n

--- a/src/doc/rustc/src/platform-support.md
+++ b/src/doc/rustc/src/platform-support.md
@@ -165,7 +165,6 @@ target | std | notes
 `i686-unknown-freebsd` | ✓ | 32-bit FreeBSD [^x86_32-floats-return-ABI]
 `i686-unknown-linux-musl` | ✓ | 32-bit Linux with musl 1.2.3 [^x86_32-floats-return-ABI]
 [`i686-unknown-uefi`](platform-support/unknown-uefi.md) | * | 32-bit UEFI
-[`loongarch64-unknown-linux-musl`](platform-support/loongarch-linux.md) | ? |  | LoongArch64 Linux (LP64D ABI) with musl 1.2.3
 [`loongarch64-unknown-none`](platform-support/loongarch-none.md) | * |  | LoongArch64 Bare-metal (LP64D ABI)
 [`loongarch64-unknown-none-softfloat`](platform-support/loongarch-none.md) | * |  | LoongArch64 Bare-metal (LP64S ABI)
 [`nvptx64-nvidia-cuda`](platform-support/nvptx64-nvidia-cuda.md) | * | --emit=asm generates PTX code that [runs on NVIDIA GPUs]
@@ -303,6 +302,7 @@ target | std | host | notes
 `i686-uwp-windows-msvc` | ? |  | [^x86_32-floats-return-ABI]
 [`i686-win7-windows-msvc`](platform-support/win7-windows-msvc.md) | ✓ |   | 32-bit Windows 7 support [^x86_32-floats-return-ABI]
 `i686-wrs-vxworks` | ? |  | [^x86_32-floats-return-ABI]
+[`loongarch64-unknown-linux-musl`](platform-support/loongarch-linux.md) | ? |  | LoongArch64 Linux (LP64D ABI) with musl 1.2.3
 [`m68k-unknown-linux-gnu`](platform-support/m68k-unknown-linux-gnu.md) | ? |  | Motorola 680x0 Linux
 `mips-unknown-linux-gnu` | ✓ | ✓ | MIPS Linux (kernel 4.4, glibc 2.23)
 `mips-unknown-linux-musl` | ✓ |  | MIPS Linux with musl 1.2.3

